### PR TITLE
[TA] Keyphrases with codegen plus some small fixes

### LINKED ISSUE #13901: CodeGen for KeyPhrases

### PR DESCRIPTION
**Pull request by maririos**
- KeyPhrases with codegen
- Makes `MultiLanguageInput.Id` and `MultiLanguageInput.Text` not required to follow behavior we had in TA 5.0.0
- Make sure we assigned the default `CountryHint` and `Language` only in the `ConvertXX` methods so we guarantee that all calls have a `CountryHint` or `Language` value even if not specified by the user.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/13901

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 1]
```diff
@@ -1601,13 +1595,13 @@ private List<TextDocumentInput> ConvertToDocumentInputs(IEnumerable<string> docu
             => documents.Select((document, i) => ConvertToDocumentInput(document, language, i)).ToList();
 
         private MultiLanguageInput ConvertToMultiLanguageInput(string document, string language, int id = 0)
-            => new MultiLanguageInput($"{id}", document) { Language = language };
+            => new MultiLanguageInput($"{id}", document) { Language = language ?? _options.DefaultLanguage};
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1598]: having defaults only in this methods

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs [hunk 2]
```diff
@@ -423,6 +423,29 @@ public async Task AnalyzeSentimentBatchWithErrorTest()
             Assert.AreEqual(exceptionMessage, ex.Message);
         }
 
+        [Test]
+        public void AnalyzeSentimentBatchWithNullIdTestAsync()
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs:427]: Adding test missed in last PR to make sure that `MultiLanguageInput` doesn't introduce a breaking change

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 3]
```diff
@@ -1171,8 +1153,8 @@ public virtual async Task<Response<ExtractKeyPhrasesResultCollection>> ExtractKe
         public virtual Response<ExtractKeyPhrasesResultCollection> ExtractKeyPhrasesBatch(IEnumerable<string> documents, string language = default, TextAnalyticsRequestOptions options = default, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(documents, nameof(documents));
-            List<TextDocumentInput> documentInputs = ConvertToDocumentInputs(documents, language);
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1174]: Missed in previous iteration, so fixing here

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 4]
```diff
@@ -1253,20 +1214,53 @@ public virtual Response<ExtractKeyPhrasesResultCollection> ExtractKeyPhrasesBatc
...
                 {
                     case 200:
-                        IDictionary<string, int> map = CreateIdToIndexMap(documents);
-                        ExtractKeyPhrasesResultCollection results = CreateKeyPhraseResponse(response, map);
+                        IDictionary<string, int> map = CreateIdToIndexMap(batchInput.Documents);
+                        ExtractKeyPhrasesResultCollection results = Transforms.ConvertToExtractKeyPhrasesResultCollection(result.Value, map);
+                        return Response.FromValue(results, response);
+                    default:
+                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false);
+                }
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1240]: `KeyPhrasesAsync` already throws if the response status is not 200, right? This `switch` may be unnecessary.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 5]
```diff
@@ -1253,20 +1214,53 @@ public virtual Response<ExtractKeyPhrasesResultCollection> ExtractKeyPhrasesBatc
...
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(ExtractKeyPhrasesBatch)}");
+            scope.Start();
+
+            try
+            {
+                Response<KeyPhraseResult> result = _serviceRestClient.KeyPhrases(batchInput, options.ModelVersion, options.IncludeStatistics, cancellationToken);
+                var response = result.GetRawResponse();
+
+                switch (response.Status)
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1259]: Same here.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs (hunk 2 above)
Comment (kinelski) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs:427]: Why the `Async` suffix in test names? Other tests don't seem to have it.

**Review by kinelski — APPROVED** [review]
LGTM. Few comments.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs]: thank you so much for catching this! you are right. We no longer need this.
I modified it for KeyPhrases and created issue https://github.com/Azure/azure-sdk-for-net/issues/14442 to do the same in all other APIs

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: thank you! it was my bad. removed Async from the test name

**Pull request merged**